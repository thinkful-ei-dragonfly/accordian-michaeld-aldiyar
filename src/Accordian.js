/* eslint-disable react/require-render-return */
/* eslint-disable no-useless-constructor */
import React from 'react';


export default class Accordian extends React.Component {
  constructor(props) {
    super(props);
  }


  render() {
    const buttons= this.props.sections.map((section, index) => (
      <button key={index}>
        {section.title}
      </button>

    ))
    return (
      <ul>
        {buttons}
      </ul>
    )
  }
}
