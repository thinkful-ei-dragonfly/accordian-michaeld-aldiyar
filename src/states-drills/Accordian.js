/* eslint-disable react/require-render-return */
/* eslint-disable no-useless-constructor */
import React from 'react'

export default class Accordian extends React.Component {
  static defaultProps = { sections: [] }
  constructor (props) {
    super(props)
    this.state = {
      activeSectionIndex: null
    }
  }

  handleButtonClick = (index) => {
    this.setState({
      activeSectionIndex: index,
    })
  }
  renderButtons () {
    return this.props.sections.map((section, index) => (
      <li key={index}>
        <button className={section.title} onClick={() => this.handleButtonClick(index)}>
          {section.title}
        </button>
        {this.state.activeSectionIndex === index && <p>{section.content}</p>}
        
      </li>
    ))
  }

  

  render () {
    return <ul>{this.renderButtons()}</ul>
  }
}
