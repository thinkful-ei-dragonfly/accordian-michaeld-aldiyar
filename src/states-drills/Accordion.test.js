import React from 'react';
import ReactDOM from 'react-dom';
import Accordian from './Accordion';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const sections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]

describe(`Accordian Component`, () => {
  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordian sections={sections} />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders empty <li>', () => {
    const tree = renderer.create(<Accordian />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('component renders no sections as active by def', () => {
    const tree = renderer.create(<Accordian sections={sections}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('component opens a clicked section', () => {
    const wrapper = shallow(<Accordian sections={sections}/>)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('component only opens the last section when mult section clicked', () => {
    const wrapper = shallow(<Accordian sections={sections}/>)
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})