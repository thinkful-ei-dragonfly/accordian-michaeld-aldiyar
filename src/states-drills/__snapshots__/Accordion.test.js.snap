// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Component component only opens the last section when mult section clicked 1`] = `
<ul
  className="Accordion"
>
   
  <li
    className="Accordion__item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="Accordion__item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    className="Accordion__item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordian Component component opens a clicked section 1`] = `
<ul
  className="Accordion"
>
   
  <li
    className="Accordion__item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="Accordion__item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    className="Accordion__item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordian Component component renders no sections as active by def 1`] = `
<ul
  className="Accordion"
>
   
  <li
    className="Accordion__item"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="Accordion__item"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    className="Accordion__item"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordian Component renders empty <li> 1`] = `
<ul
  className="Accordion"
>
   
  <li />
</ul>
`;
